apps/sms/views/conversation/test/unit/attachment_test.js

```diff
@@ -145,62 +145,67 @@ suite('attachment_test.js', function() {
       var attachment = new Attachment(testImageBlob, {
         name: fileName
       });
       var typeSpy = MimeMapper.guessTypeFromFileProperties;
       var matchSpy = MimeMapper.ensureFilenameMatchesType;
-      attachment.view();
-      assert.ok(typeSpy.calledWith(fileName, 'image/jpeg'));
-      assert.ok(matchSpy.calledWith(fileName, typeSpy.returnValues[0]));
-      assert.equal(MockMozActivity.calls.length, 1);
+      attachment.view().then(() => {
+        assert.ok(typeSpy.calledWith(fileName, 'image/jpeg'));
+        assert.ok(matchSpy.calledWith(fileName, typeSpy.returnValues[0]));
+        assert.equal(MockMozActivity.calls.length, 1);
+      }).then(done, done);
     });
 
-    test('Filename has no extension', function() {
+    test('Filename has no extension', function(done) {
       var fileName = 'jpeg_image.jpg';
       var attachment = new Attachment(testImageBlob, {
         name: fileName
       });
       var typeSpy = MimeMapper.guessTypeFromFileProperties;
       var matchSpy = MimeMapper.ensureFilenameMatchesType;
-      attachment.view();
-      assert.ok(typeSpy.calledWith(fileName, 'image/jpeg'));
-      assert.ok(matchSpy.calledWith(fileName, typeSpy.returnValues[0]));
-      assert.equal(MockMozActivity.calls.length, 1);
+      attachment.view().then(() => {
+        assert.ok(typeSpy.calledWith(fileName, 'image/jpeg'));
+        assert.ok(matchSpy.calledWith(fileName, typeSpy.returnValues[0]));
+        assert.equal(MockMozActivity.calls.length, 1);
+      }).then(done, done);
     });
 
-    test('Filename is overridden using single attachment folder', function() {
+    test('Filename is overridden using single attachment folder',
+      function(done) {
+
       var attachment1 = new Attachment(testImageBlob, {
         name: '/some_path/.hidden_folder/attachment1.jpg'
       });
 
-      attachment1.view();
-
-      assert.equal(MockMozActivity.calls.length, 1);
-      assert.equal(
-        MockMozActivity.calls[0].data.filename,
-        'sms-attachments/attachment1.jpg'
-      );
+      attachment1.view().then(() => {
 
-      var attachment2 = new Attachment(testImageBlob, {
-        name: 'attachment2.jpg'
-      });
+        assert.equal(MockMozActivity.calls.length, 1);
+        assert.equal(
+          MockMozActivity.calls[0].data.filename,
+          'sms-attachments/attachment1.jpg'
+        );
 
-      attachment2.view();
+        var attachment2 = new Attachment(testImageBlob, {
+          name: 'attachment2.jpg'
+        });
 
-      assert.equal(MockMozActivity.calls.length, 2);
-      assert.equal(
-        MockMozActivity.calls[1].data.filename,
-        'sms-attachments/attachment2.jpg'
-      );
+        return attachment2.view();
+      }).then(() => {
+        assert.equal(MockMozActivity.calls.length, 2);
+        assert.equal(
+          MockMozActivity.calls[1].data.filename,
+          'sms-attachments/attachment2.jpg'
+        );
+      }).then(done, done);
 
     });
```

Comment:
Could you add one more test about the empty file name case to verify your change for the attachment?